model/trigger/build.go

```diff
@@ -208,3 +82,135 @@ func taskStatusSubformat(n int, verb string) string {
 func appendTime(b *build.Build, txt string) string {
 	return fmt.Sprintf("%s in %s", txt, b.FinishTime.Sub(b.StartTime).String())
 }
+
+type buildTriggers struct {
+	event    *event.EventLogEntry
+	data     *event.BuildEventData
+	build    *build.Build
+	uiConfig evergreen.UIConfig
+
+	base
+}
+
+func makeBuildTriggers() eventHandler {
+	t := &buildTriggers{}
+	t.base.triggers = map[string]trigger{
+		triggerOutcome: t.buildOutcome,
+		triggerFailure: t.buildFailure,
+		triggerSuccess: t.buildSuccess,
+	}
+	return t
+}
+
+func (t *buildTriggers) Fetch(e *event.EventLogEntry) error {
+	var err error
+	if err = t.uiConfig.Get(); err != nil {
+		return errors.Wrap(err, "Failed to fetch ui config")
+	}
+
+	t.build, err = build.FindOne(build.ById(e.ResourceId))
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch build")
+	}
+	if t.build == nil {
+		return errors.New("couldn't find build")
+	}
+
+	var ok bool
+	t.data, ok = e.Data.(*event.BuildEventData)
+	if !ok {
+		return errors.Wrapf(err, "build '%s' contains unexpected data with type '%T'", e.ResourceId, e.Data)
+	}
+	t.event = e
+
+	return nil
+}
+
+func (t *buildTriggers) Selectors() []event.Selector {
+	return []event.Selector{
+		{
+			Type: selectorID,
+			Data: t.build.Id,
+		},
+		{
+			Type: selectorObject,
+			Data: objectBuild,
+		},
+		{
+			Type: selectorProject,
+			Data: t.build.Project,
+		},
+		{
+			Type: selectorRequester,
+			Data: t.build.Requester,
+		},
+		{
+			Type: selectorInVersion,
+			Data: t.build.Version,
+		},
+	}
+}
+
+func (t *buildTriggers) buildOutcome(sub *event.Subscription) (*notification.Notification, error) {
+	if t.data.Status != evergreen.BuildSucceeded && t.data.Status != evergreen.BuildFailed {
+		return nil, nil
+	}
+
+	return t.generate(sub)
+}
+
+func (t *buildTriggers) buildFailure(sub *event.Subscription) (*notification.Notification, error) {
+	if t.data.Status != evergreen.BuildFailed {
+		return nil, nil
+	}
+
+	return t.generate(sub)
+}
+
+func (t *buildTriggers) buildSuccess(sub *event.Subscription) (*notification.Notification, error) {
+	if t.data.Status != evergreen.BuildSucceeded {
+		return nil, nil
+	}
+
+	return t.generate(sub)
+}
+
+func (t *buildTriggers) makeData(sub *event.Subscription) (*commonTemplateData, error) {
+	api := restModel.APIBuild{}
```

Comment:
The db model can't have a dependency on the rest model. We may need a rest version of the commonTemplateData

Reply:
There is no circular dependency at the moment, so it's fine. There is another ticket to move this out of the model package.